src/components/Modals/AddContactModal.jsx

```diff
@@ -48,59 +47,45 @@ const AddContactModal = ({ addContact, showAddContactModal, setShowAddContactMod
   const { addNotification } = useNotification();
   const [userGivenName, setUserGivenName] = useState('');
   const [userFamilyName, setUserFamilyName] = useState('');
-  const [username, setUsername] = useState('');
   const [webId, setWebId] = useState('');
   const [invalidWebId, setInvalidWebId] = useState(false);
   const [processing, setProcessing] = useState(false);
   const theme = useTheme();
   const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
 
-  const wrappedSetUsername = (value) => {
-    setUsername(value);
-    const renderedWebId = renderWebId(value);
-    setWebId(renderedWebId);
-    if (invalidWebId) setInvalidWebId(false);
-  };
-
   const clearInputFields = () => {
     setUserGivenName('');
     setUserFamilyName('');
-    setUsername('');
     setWebId('');
     setInvalidWebId(false);
   };
 
   const handleAddContact = async (event) => {
     event.preventDefault();
     setProcessing(true);
+
     const { addUserGivenName, addUserFamilyName, addWebId } = event.target.elements;
+
     const userObject = {
-      givenName: addUserGivenName.value.trim(),
-      familyName: addUserFamilyName.value.trim(),
-      webId: addWebId.value.trim()
+      webId: addWebId.value.trim(),
+      ...(addUserGivenName.value.trim() && { givenName: addUserGivenName.value.trim() }),
+      ...(addUserFamilyName.value.trim() && { familyName: addUserFamilyName.value.trim() })
     };
 
-    // Validation for webId
     try {
       await getWebIdDataset(userObject.webId);
-    } catch {
-      addNotification('error', `Add contact failed. Reason: ${userObject.webId} does not exist`);
-      setInvalidWebId(true);
-      setProcessing(false);
-      return;
-    }
-
-    try {
       await addContact(userObject);
-      addNotification(
-        'success',
-        `"${userObject.givenName} ${userObject.familyName}" added to contact list`
-      );
+      const nameDisplay =
+        [userObject.givenName, userObject.familyName].filter(Boolean).join(' ') || 'Contact';
```

Comment:
We can use the `webId` instead of a generic "Contact" as `nameDisplay`

Reply:
Thanks! I'm happy to get to work addressing these. Before I do, I just want to confirm, you do want to re-introduce username? as an optional field. I know the issue was straightforward about wanting it removed. There definitely some quirks going on here for sure.

Reply:
also, i've gotten the delete confirmation and notification to display the webid instead of the person, but would we want maybe to program some logic so that the messages display a name or combo of names if there are any, but will display web id if there are no names? or are we happy with it just displaying the webId?

Reply:
> Thanks! I'm happy to get to work addressing these. Before I do, I just want to confirm, you do want to re-introduce username? as an optional field. I know the issue was straightforward about wanting it removed. There definitely some quirks going on here for sure.

Yeah, I think it'll be beneficial as an optional field. But we could leave leave this removed if we want to follow the issue.

Reply:
> also, i've gotten the delete confirmation and notification to display the webid instead of the person, but would we want maybe to program some logic so that the messages display a name or combo of names if there are any, but will display web id if there are no names? or are we happy with it just displaying the webId?

Yes, that's correct. Either the webid when there's no names available or just the name themselves when they are.

Reply:
okay, i've pushed the changes, and I believe it is all addressed and taken care of.

I did leave the username field out for now, as I think we should go over this section in a meeting a get a consensus, or at least get wilfreds input, as it was his issue I believe. There are some ux/ui developments that could still be made in the section as a whole, would be good to workshop it. 

Lastly, I'm noticing the test is failing still, I'm curious what is going on, but I don't have much experience with running these kinds of tests. Would love to know what I should do differently, or what I can learn from this, thanks!

Reply:
Because of the the functional and component changes, the unit tests are failing since it's likely testing for things that were removed or altered.

I recommend starting with the tests files with the same name as the components and hooks and look/read up on tutorials or reaourcess for basic unit tests with jest or vitest like https://medium.com/@masbagaspn/unit-testing-react-application-with-vitest-and-react-testing-library-910f6f4dc675 

Think that should give you a head start

Reply:
okay. I believe I've correctly rewritten the tests based on the changed functionality implemented in the contact list, but I am new to this process so let me know if everything looks good. thanks!